internal/retry/config_test.go

```diff
@@ -0,0 +1,408 @@
+					BackoffStrategy(
+						exponentialNoError(backoff.NewExponential(
+							backoff.FirstBackoff(time.Millisecond*10),
+							backoff.MaxBackoff(time.Second*1),
+						)),
+					),
+				),
+			).provider,
+		},
+		{
+			msg: "invalid default policy",
+			retryConfig: whitespace.Expand(`
+				policies:
+					realpolicy:
+						retries: 1
+						maxtimeout: 500ms
+						backoff:
+							exponential:
+								first: 10ms
+								max: 1s
+				default: fakepolicy
+			`),
+			wantError: []string{
+				`invalid default retry policy: "fakepolicy", possiblities are:`,
+				`realpolicy`,
+			},
+		},
+		{
+			msg: "invalid override policy",
+			retryConfig: whitespace.Expand(`
+				policies:
+					realpolicy:
+						retries: 1
+						maxtimeout: 500ms
+						backoff:
+							exponential:
+								first: 10ms
+								max: 1s
+				default: realpolicy
+				overrides:
+					- service: myservice
+					  with: fakepolicy
+			`),
+			wantError: []string{
+				`invalid retry policy: "fakepolicy", possiblities are:`,
+				`realpolicy`,
+			},
+		},
+		{
+			msg: "missing with in override policy",
+			retryConfig: whitespace.Expand(`
+				policies:
+					realpolicy:
+						retries: 1
+						maxtimeout: 500ms
+						backoff:
+							exponential:
+								first: 10ms
+								max: 1s
+				default: realpolicy
+				overrides:
+					- service: myservice
+			`),
+			wantError: []string{
+				`invalid retry policy: "", possiblities are:`,
+				`realpolicy`,
+			},
+		},
+		{
+			msg: "no service or procedure in override policy",
+			retryConfig: whitespace.Expand(`
+				policies:
+					realpolicy:
+						retries: 1
+						maxtimeout: 500ms
+						backoff:
+							exponential:
+								first: 10ms
+								max: 1s
+				default: realpolicy
+				overrides:
+					- with: realpolicy
+			`),
+			wantError: []string{
+				`did not specify a service or procedure for retry policy override: "realpolicy"`,
+			},
+		},
+		{
+			msg: "invalid policy retries",
+			retryConfig: whitespace.Expand(`
+				policies:
+					realpolicy:
+						retries: -1
+						maxtimeout: 500ms
+						backoff:
+							exponential:
+								first: 10ms
+								max: 1s
+				default: realpolicy
+			`),
+			wantError: []string{
+				`cannot parse`,
+				`realpolicy`,
+			},
+		},
+		{
+			msg: "invalid policy timeout",
+			retryConfig: whitespace.Expand(`
+				policies:
+					realpolicy:
+						retries: 1
+						maxtimeout: abc
+						backoff:
+							exponential:
+								first: 10ms
+								max: 1s
+				default: realpolicy
+			`),
+			wantError: []string{
+				`error decoding`,
+				`realpolicy`,
+			},
+		},
+		{
+			msg: "invalid policy backoff",
+			retryConfig: whitespace.Expand(`
+				policies:
+					realpolicy:
+						retries: 1
+						maxtimeout: abc
+						backoff: error
+				default: realpolicy
+			`),
+			wantError: []string{
+				`error decoding`,
+				`realpolicy`,
+			},
+		},
+		{
+			msg:                "fallsback to default policy",
+			retryConfig:        whitespace.Expand(``),
```

Comment:
Let’s generally move Expand into the test runner. This line can be omitted entirely since it’s a null string.